### PR TITLE
592 create redirect pages within loop

### LINKED ISSUE #592: Create Redirect pages within Loop
Staff are interested in having the Redirect Page type from the Wagtail public site available on Loop, as well. Can this be added there?

### PR DESCRIPTION
**Pull request by bluairis**
Fixes #592

Creates a redirect page type for loop pages that generates sidebar links.

# Testing instructions for creating a sidebar link:
1. Navigate to Loop > Groups in the wagtail admin interface
2. Create a child page of one of the pages in Groups
3. Select Loop Redirect Page from the menu of options
4. Enter the link text under title and the URL under redirect URL
5. Populate the rest of the required fields in the content tab
6. Navigate to the promote tab and check the "Show in menus" checkbox under "for site menus"
7. Publish the page

### CODE REVIEW

File: redirects/models.py [hunk 1]
```diff
@@ -31,3 +31,29 @@ def serve(self, request):
         Override the serve method to create a redirect. 
         """
         return redirect(self.link, permanent=True)
+
```
Comment (bbusenius) [inline redirects/models.py:34]: Extra blank line should be here.

File: redirects/models.py [hunk 2]
```diff
@@ -31,3 +31,29 @@ def serve(self, request):
...
+            [
+                FieldPanel('redirect_url'),
+                PageChooserPanel('link_page')
+            ]
+        )
+    ] + BasePage.content_panels
+
+    def serve(self, request):
+        """
+        Override the serve method to create a redirect.
```
Comment (bbusenius) [inline redirects/models.py:57]: Delete space at end of line.

File: redirects/models.py [hunk 3]
```diff
@@ -31,3 +31,29 @@ def serve(self, request):
...
+class LoopRedirectPage(BasePage, LinkFields):
+    """
+    Page object for setting up redirects in loop that need to 
+    appear in the sidebar in a given section of the site.
+    """
+    subpage_types = []
+
+    redirect_url = models.URLField(
+        max_length=200,
+        default = "https://loop.lib.uchicago.edu/")
```
Comment (bbusenius) [inline redirects/models.py:44]: Remove spaces around `=`.

**Review by bbusenius — CHANGES_REQUESTED** [review]
This looks good. Many of the changes I requested are _very_ minor coding style complaints that show up in my editor. These should be showing up in your editor too. If they aren't showing up there, we might need to do some setup with whatever editor you're using. I can help you with that later but if you just want to manually make the changes now, that's fine.

I know you're aware of this and have been waiting for the new Vagrant environment but just as a reminder, you will need to add the following to this pull request:
1. Migrations
2. Fixtures

To help you with that, I will merge master into this branch. Then you should be able to spin it up in the dev environment.

**Comment by bufordrat** [comment]
Doh!  How did I forget about the fixtures?  Thanks, Brad.  Will help Irina w/ that next time she's in.

File: redirects/models.py [hunk 4]
```diff
@@ -4,7 +4,7 @@
 from wagtail.admin.edit_handlers import FieldPanel, FieldRowPanel, MultiFieldPanel, PageChooserPanel
 from wagtail.documents.edit_handlers import DocumentChooserPanel
 from wagtail.search import index
-from base.models import PublicBasePage, LinkFields
+from base.models import PublicBasePage, LinkFields, BasePage
```
[the remark this answers is not in the document]
Comment (bluairis) [reply redirects/models.py:8]: When I isort it re-arranges the imports to:
(where `MultiFieldPanel, PageChooserPanel)` is indented)

`from base.models import BasePage, LinkFields, PublicBasePage`
`from django.db import models`
`from django.shortcuts import redirect`
`from wagtail.admin.edit_handlers import (FieldPanel, FieldRowPanel,`
`MultiFieldPanel, PageChooserPanel)`

`from wagtail.core.models import Page`
`from wagtail.documents.edit_handlers import DocumentChooserPanel`
`from wagtail.search import index'

**Review by bbusenius — APPROVED** [review]
Looks good! Thank you!

**Review by bufordrat — APPROVED** [review]
Yup, looks good; ready to go now that we have the fixtures and migrations.

**Pull request merged**